src/driver/hermes_mod.c

```diff
@@ -55,14 +55,74 @@ static const struct pci_device_id pci_ids[] = {
 };
 MODULE_DEVICE_TABLE(pci, pci_ids);
 
+struct ebpf_irq_arg {
+	struct hermes_dev *hdev;
+	int eng;
+} ebpf_irq_args[EBPF_ENG_NUM_MAX];
```

Comment:
This global variable seems highly questionable... What happens if there are multiple devices in the system?

Should really have a per-engine structure that's allocated at init for this kind of thing. Then there's no need for the silly MAX define.

Reply:
Yes... that makes total sense...